server/executor/assertion_runner.go

```diff
@@ -122,7 +123,12 @@ func (e *defaultAssertionRunner) runAssertionsAndUpdateResult(ctx context.Contex
 			log.Printf("[AssertionRunner] Test %s Run %d: fail to emit TestSpecsRunError event: %s\n", request.Test.ID, request.Run.ID, anotherErr.Error())
 		}
 
-		return model.Run{}, e.updater.Update(ctx, run.AssertionFailed(err))
+		run = run.AssertionFailed(err)
```

Comment:
I was trying to find a single place where we finished a test run no matter the outcome, but I didn't find it, it feels that wherever the error is found we are just updating the required fields.

I know that the `run.go` model has a `Finish` method but it didn't feel right to execute an analytics event from a model. 

We need to consider having a single spot to finish a test run that is not the model where we could execute side effects in the future, and not having to scatter the logic across these files.